errors/OECreationFailed.h

```diff
@@ -0,0 +1,13 @@
+#ifndef OECREATION_H
```

Comment:
Surely we can just have one "errors" header that defines many error classes right?